homeassistant/components/tasmota_irhvac/climate.py

```diff
@@ -0,0 +1,695 @@
+        """Set new target fan mode."""
+        if fan_mode not in self._fan_list:
+            _LOGGER.error(
+                "Invalid swing mode selected. Got '%s'. Allowed modes are:", fan_mode
+            )
+            _LOGGER.error(self._fan_list)
+            return
+        self._fan_mode = fan_mode
+        self.power_mode = STATE_ON
+        await self.hass.async_add_executor_job(self.send_ir)
+        await self.async_update_ha_state()
+
+    async def async_set_swing_mode(self, swing_mode):
+        """Set new target swing operation."""
+        if swing_mode not in self._swing_list:
+            _LOGGER.error(
+                "Invalid swing mode selected. Got '%s'. Allowed modes are:", swing_mode
+            )
+            _LOGGER.error(self._swing_list)
+            return
+        self._swing_mode = swing_mode
+        self.power_mode = STATE_ON
+        await self.hass.async_add_executor_job(self.send_ir)
+        await self.async_update_ha_state()
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        if self._min_temp:
+            return self._min_temp
+
+        # get default temp from super class
+        return super().min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature."""
+        if self._max_temp:
+            return self._max_temp
+
+        # Get default temp from super class
+        return super().max_temp
+
+    async def _async_sensor_changed(self, entity_id, old_state, new_state):
+        """Handle temperature changes."""
+        if new_state is None:
+            return
+
+        self._async_update_temp(new_state)
+        await self.async_update_ha_state()
+
+    @callback
+    def _async_update_temp(self, state):
+        """Update thermostat with latest state from sensor."""
+        try:
+            self._cur_temp = float(state.state)
+        except ValueError as ex:
+            _LOGGER.error("Unable to update from sensor: %s", ex)
+
+    @property
+    def _is_device_active(self):
+        """If the toggleable device is currently active."""
+        return self.power_mode == STATE_ON
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return self._support_flags
+
+    async def async_set_preset_mode(self, preset_mode):
+        """Set new preset mode.
+
+        This method must be run in the event loop and returns a coroutine.
+        """
+        if preset_mode == PRESET_AWAY and not self._is_away:
+            self._is_away = True
+            self._saved_target_temp = self._target_temp
+            self._target_temp = self._away_temp
+        elif preset_mode == PRESET_NONE and self._is_away:
+            self._is_away = False
+            self._target_temp = self._saved_target_temp
+        await self.hass.async_add_executor_job(self.send_ir)
+        await self.async_update_ha_state()
+
+    def send_ir(self):
```

Comment:
This method should be part of the library as it contains protocol specific information.

See the mysensors integration for an example how to pass the mqtt publish callback to the 3rd party library.